### DISCUSSION TITLE
Can't use AWS credentials with my NuxtJS app

### DISCUSSION THREAD

**Discussion by ArthurLERAY** [vercel/community#1454] [Help] [answered]
### Summary

I am unable to request my AWS S3 due to credentials error.
I made sure not to use reserved environnement variables and added mine directly on the UI.
My project works fine locally.

I'm using `nuxt@^2.15.8` and `aws-sdk@^2.1295.0` for this project. (My node version is `16.17.0`)

Here is the relevant code part :
```js
import AWS from "aws-sdk";
const s3 = new AWS.S3({
        apiVersion: '2006-03-01',
        signatureVersion: 'v4',
        region: 'eu-west-3',
        credentials: {
          accessKeyId: process.env.NUXT_APP_AWS_KEY_ID,
          secretAccessKey: process.env.NUXT_APP_AWS_SECRET_KEY,
        }
      });
const images = ["image_key.jpg"];
images = images.map(i => {
        const params = {
          Bucket: process.env.NUXT_APP_AWS_BUCKET,
          Key: i
        };
        return {key: i, url: s3.getSignedUrl('getObject', params)};
    });
```

And I get an error on this `getSignedUrl` :
`CredentialsError: Missing credentials in config, if using AWS_CONFIG_FILE, set AWS_SDK_LOAD_CONFIG=1`

In my Nuxt project, I defined a plugin named `aws.js` containing :

```js
import AWS from "aws-sdk";

export default function({ app }) {
  AWS.config.update({
    accessKeyId: process.env.NUXT_APP_AWS_KEY_ID,
    secretAccessKey: process.env.NUXT_APP_AWS_SECRET_KEY
  });
}
```

and used it in my `nuxt.config.js` :
```js
plugins: [
    '~/plugins/aws.js'
  ]
```

Can someone tell me if I'm doing something wrong with my configuration ?

### Additional information

`package.json`:
```json
{
  "name": "app",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "dev": "nuxt",
    "build": "nuxt build",
    "start": "nuxt start",
    "generate": "nuxt generate"
  },
  "dependencies": {
    "@nuxt/typescript-build": "^2.1.0",
    "@nuxtjs/auth-next": "5.0.0-1667386184.dfbbb54",
    "@nuxtjs/axios": "^5.13.6",
    "@nuxtjs/dotenv": "^1.4.1",
    "@nuxtjs/sitemap": "^2.4.0",
    "@nuxtjs/toast": "^3.3.1",
    "@tailwindcss/forms": "^0.5.3",
    "@tailwindcss/line-clamp": "^0.4.2",
    "@vueform/slider": "^2.1.6",
    "aws-sdk": "^2.1295.0",
    "core-js": "^3.25.3",
    "js-sha256": "^0.9.0",
    "jscharting-vue": "^2.3.1",
    "luxon": "^3.1.1",
    "nuxt": "^2.15.8",
    "tailwindcss": "^3.2.4",
    "ts-luxon": "^4.0.1",
    "vue": "^2.7.10",
    "vue-server-renderer": "^2.7.10",
    "vue-template-compiler": "^2.7.10",
    "vue-uuid": "^3.0.0",
    "vue3-easy-data-table": "^1.5.20"
  },
  "devDependencies": {
    "@nuxtjs/tailwindcss": "^5.3.3",
    "postcss": "^8.4.17"
  },
  "engines": {
    "node": "16.x"
  }
}
```

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out, Arthur! Have you added the [Environment Variables](https://vercel.com/docs/concepts/projects/environment-variables#) in the project settings UI (`vercel.com/<username>/<project-name>/settings/environment-variables`), or did you add them a different way?

**Reply by ArthurLERAY** [reply]
Hi, yes I added them directly in the project settings UI.
Each variable is available for Production, Preview and Development environment.

I tried different names and redeploys but variables don't seem to be accessible by my app

**Reply by amyegan** [reply]
I did some digging and found a few possible causes/solutions:
- change your variable names to use the `NUXT_ENV_` prefix (source: [Nuxt.js Environment Variables (without dotenv)](https://dev.to/deeja/nuxt-js-environment-variables-without-dotenv-4oj8#automatic-raw-env-endraw-variables-using-raw-nuxtenv-endraw-))
- use [the env property](https://nuxtjs.org/docs/configuration-glossary/configuration-env/) (source: [Declared `env` variables using the `nuxt.config.js`](https://dev.to/deeja/nuxt-js-environment-variables-without-dotenv-4oj8#declared-raw-env-endraw-variables-using-the-raw-nuxtconfigjs-endraw-))
- use [Nuxt runtime config](https://nuxtjs.org/docs/configuration-glossary/configuration-runtime-config/#publicruntimeconfig) (source: https://stackoverflow.com/a/73524243 and https://stackoverflow.com/a/67705541)

**Reply by ArthurLERAY** [reply]
Hi, I renamed every env variable from NUXT_APP to NUXT_ENV and it works.

Thanks for your help !
